contrib/src/main/java/com/datatorrent/contrib/avro/AvroRecordHelper.java

```diff
@@ -114,6 +107,7 @@ private static Object convertAndResolveUnionToPrimitive(Schema schema, String ke
         }
         convertedValue = convertValueToAvroPrimitive(types.get(i).getType(), key, value);
       } catch (RuntimeException e) {
+        LOG.error("Could not handle schema resolution" + e.getMessage());
```

Comment:
Actually failure for reading a single field inside a union should not cause a failure to read the rest of the values inside the union.